services/cad-service/grpc_server.py

```
@@ -0,0 +1,93 @@
+
+def _async_iterate(iterator):
+    """Adapt a synchronous iterator to an asynchronous one."""
+
+    async def _iterate():
+        for item in iterator:
+            yield item
+
+    return _iterate()
+
+
+async def serve(host: str = "0.0.0.0", port: int = 50051) -> grpc.aio.Server:
```

Ruff (0.14.1)

Possible binding to all interfaces

(S104)